src/libraries/System.Private.CoreLib/src/System/Collections/ListDictionaryInternal.cs

```diff
@@ -202,6 +203,17 @@ public void Remove(object key)
             count--;
         }
 
+        internal KeyValuePairs[] ToKeyValuePairsArray()
+        {
+            var array = new KeyValuePairs[count];
```

Comment:
`KeyValuePairs` is an unusual type, and the debugger output differs from other dictionaries. It looks like only `Hashtable` uses it, and every other `IDictionary` uses the generic `KeyValuePair<object, object>`.

I think this should use `KeyValuePair<object, object>` to be consistent with the rest of the ecosystem.

Reply:
This method is only used by the debug view. Can it be moved into the debug view?

Reply:
This can just be the implementation of the Items property in the debug view; doesn't need its own method.

Reply:
I based my changes on two other non-generic collection classes, `Hashtable` and `SortedList`. To keep code and behaviour consistent with existing examples I used a similar code structure and item type.

Personally, I prefer the way how `KeyValuePairs` presents items (it does it using an additional parameter of `DebuggerDisplay`) but I agree that it would be nice to have greater consistency between generic and non-generic collections.

Maybe instead of modifying this PR, it would be better to make suggested refactorings on all mentioned classes in a separate PR? Otherwise, inconsistencies in behaviour and code between collection classes will remain. But if you think that it is important to modify this PR please let me know and I'll update it.

Reply:
@JamesNK, @stephentoub I inlined the method as suggested. Earlier I incorrectly interpreted why that method was needed in other places.

@JamesNK I created #91506 to discuss how to display dictionaries.